app/src/main/java/com/concordia/mcga/adapters/POISearchAdapter.java

```diff
@@ -0,0 +1,283 @@
+    @Override
+    public Object getChild(int i, int i1) {
+        Campus camp = (Campus)getGroup(i);
+        if (camp == Campus.SGW) {
+            return sgwFilteredList.get(i1);
+        } else if (camp == Campus.LOY) {
+            return loyolaFilteredList.get(i1);
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public long getGroupId(int i) {
+        return i;
+    }
+
+    @Override
+    public long getChildId(int i, int i1) {
+        return i1;
+    }
+
+    @Override
+    public boolean hasStableIds() {
+        return true;
+    }
+
+    /**
+     * Helper function to determine whether a given filtered POI list (from the query) is empty
+     * @param index The campus to query
+     * @return True if the query-filtered campus list is empty, false if it has items
+     */
+    public boolean getGroupIsEmpty(int index) {
+        if (index == SGW_INDEX) {
+            return sgwFilteredList.isEmpty();
+        } else if (index == LOYOLA_INDEX) {
+            return loyolaFilteredList.isEmpty();
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Inflates and returns the view associated with the index-specified campus or POI group
+     * @param i Campus/group index
+     * @param b Whether the group is collapsed or expanded
+     * @param view The old view to reuse, if possible
+     * @param viewGroup The parent that this view will be attached to
+     * @return The inflated POI group view
+     */
+    @Override
+    public View getGroupView(int i, boolean b, View view, ViewGroup viewGroup) {
+        Campus campus = (Campus)getGroup(i);
+
+        if (campus == null) {
+            return null;
+        }
+
+        if (view == null) {
+            LayoutInflater layoutInflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+            view = layoutInflater.inflate(R.layout.poi_search_group_row, null);
+        }
+
+        CircleImageView campusImage = (CircleImageView) view.findViewById(R.id.groupImage);
+        Campus camp = (Campus)getGroup(i);
+        if (camp == Campus.SGW) {
+            campusImage.setImageResource(R.mipmap.ic_sgw_campus);
+        } else {
+            campusImage.setImageResource(R.mipmap.ic_loy_campus);
+        }
+
+        TextView campusRow = (TextView) view.findViewById(R.id.groupText);
+        campusRow.setText(campus.getName());
+
+        return view;
+    }
+
+    /**
+     * Inflates and returns the view associated with the POI list item. For now, a building name
+     * @param i The group/campus index
+     * @param i1 The child/POI index
+     * @param b Whether the child is the last one in the group
+     * @param view Existing view to reuse if possible
+     * @param viewGroup The parent group this view will be attached to
+     * @return The POI menu item (building name)
+     */
+    @Override
+    public View getChildView(int i, int i1, boolean b, View view, ViewGroup viewGroup) {
```

Comment:
`groupPosition` `childPosition` I think it makes more sense when overriding a method to keep the parameter names the same as in the base class, at least for the ones that are actually used.